src/front/macOS/kDriveCore/ServerBridge/XPC/NSXPCConnection+SwiftConcurrency.swift

```diff
@@ -0,0 +1,44 @@
+/*
+ Infomaniak kDrive - Desktop
+ Copyright (C) 2023-2025 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+
+public extension NSXPCConnection {
+    /// Async/Await wrapping of remoteObjectProxyWithErrorHandler
+    func asyncProxy<Interface>(
```

Comment:
Maybe we can come with a better name more explicit about what the function does?